src/NetMQ.Tests/ClientServer.cs

```diff
@@ -71,7 +79,64 @@ public async void AsyncWithCancellationToken()
             
             source.CancelAfter(100);
 
-            await Assert.ThrowsAsync<OperationCanceledException>(async () => await server.ReceiveStringAsync(source.Token));
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await server.ReceiveStringAsync(source.Token));
         }
+
+#if NETCOREAPP3_1
+        
+        [Fact]
+        public void AsyncEnumerable()
+        {
+            using var server = new ServerSocket();
+            int port = server.BindRandomPort("tcp://*");
+            
+            using var client = new ClientSocket();
+            client.Connect($"tcp://127.0.0.1:{port}");
+
+            var t1 = Task.Run(async () =>
+            {
+                int count = 0;
+                
+                await foreach (var (_, msg) in server.ReceiveStringAsyncEnumerable())
+                {
+                    count++;
+
+                    if (msg == "1")
+                    {
+                        m_testOutputHelper.WriteLine($"T1 read {count} messages");
+                        return;
+                    }
+                }
+            });
+            
+            var t2 = Task.Run(async () =>  {
+                int count = 0;
+                
+                await foreach (var (_, msg) in server.ReceiveStringAsyncEnumerable())
+                {
+                    count++;
+
+                    if (msg == "1")
+                    {
+                        m_testOutputHelper.WriteLine($"T2 read {count} messages");
+                        return;
+                    }
+                }
+            });
+
+            for (int i = 0; i < 15000; i++)
+            {
+                client.Send("0");
+            }
+            
+            // Send the end message to both of the threads
+            client.Send("1");
+            client.Send("1");
+
+            t1.Wait();
+            t2.Wait();
```

Comment:
nit: If this test could deadlock then awaiting these tasks with a timeout would be good.